libbeat/metric/system/diskio/diskstat_linux.go

```diff
@@ -52,6 +53,28 @@ func (stat *IOStat) OpenSampling() error {
 	return stat.curCPU.Get()
 }
 
+// a few of the diskio counters are actually 32-bit on the kernel side, which means they can roll over fairly easily.
+// Here we try to reconstruct the values by calculating the pre-rollover delta from unt32 max, then adding.
+// If you want to get technical, this could be a tad unsafe, as we don't actually have any way of knowing if the word size changes in a future kernel, and we've rolled over at UINT64_MAX
+
+// See https://docs.kernel.org/admin-guide/iostats.html and https://github.com/torvalds/linux/blob/master/block/genhd.c
+func returnOrFix(current, prev uint64) uint64 {
+	var maxUint32 uint64 = 4_294_967_295 // Max value in uint32/unsigned int
```

Comment:
The `math` package has this as a constant if you want: https://pkg.go.dev/math#pkg-constants: `math.MaxUint32`

Reply:
++ for using the constant in the `math` package. You can also add the value as a comment, if that facilitates reading the code.